langgenius/dify
# Feat: Add Top bar while routing different different pages

# Title: Implement Dynamic Loading Indicator Using Next-NProgress

### Overview
To enhance user experience during page loads on both self-hosted and cloud versions of our app, I've integrated the next-nprogress-bar. This feature provides a visual loading indicator, improving the perceived performance and responsiveness.

### Changes

- Added next-nprogress-bar for a top-loading progress indicator.
- Configured the progress bar to match our app's loading dynamics and style.

### Testing Steps
- Check out the feature branch.
- Run npm install to update dependencies.
- Start the app and navigate between pages to see the progress bar in action.

### Feedback Request
Please share feedback on the progress bar's look and feel, any observed performance impacts, and suggestions for adjustments.
